great_expectations/profile/user_configurable_profiler.py

```diff
@@ -0,0 +1,1071 @@
+                        ],
+                    },
+                    allow_relative_error=allow_relative_error,
+                    catch_exceptions=True,
+                )
+                dataset.set_config_value("interactive_evaluation", True)
+        return dataset
+
+    def _build_expectations_primary_or_compound_key(
+        self, dataset, column_list, **kwargs
+    ):
+        """
+        Adds a uniqueness expectation for a given column or set of columns
+        Args:
+            dataset: A GE Dataset
+            column_list: A list containing one or more columns for which to add a uniqueness expectation
+            **kwargs:
+
+        Returns:
+            The GE Dataset
+        """
+        # uniqueness
+        if (
+            len(column_list) > 1
+            and "expect_compound_columns_to_be_unique" not in self.excluded_expectations
+        ):
+            dataset.expect_compound_columns_to_be_unique(column_list)
+        elif len(column_list) < 1:
+            raise ValueError(
+                "When specifying a primary or compound key, column_list must not be empty"
+            )
+        else:
+            [column] = column_list
+            if "expect_column_values_to_be_unique" not in self.excluded_expectations:
+                dataset.expect_column_values_to_be_unique(column)
+        return dataset
+
+    def _build_expectations_string(self, dataset, column, **kwargs):
+        """
+        Adds a set of string expectations for a given column. Currently does not do anything.
+        With the 0.12 API there isn't a quick way to introspect for value_lengths - if we did that, we could build a
+        potentially useful value_lengths expectation here.
+        Args:
+            dataset: A GE Dataset
+            column: The column for which to add expectations
+            **kwargs:
+
+        Returns:
+            The GE Dataset
+        """
+
+        if (
+            "expect_column_value_lengths_to_be_between"
+            not in self.excluded_expectations
+        ):
+
+            pass
+        return dataset
+
+    def _build_expectations_datetime(self, dataset, column, **kwargs):
+        """
+        Adds `expect_column_values_to_be_between` for a given column
+        Args:
+            dataset: A GE Dataset
+            column: The column for which to add the expectation
+            **kwargs:
+
+        Returns:
+            The GE Dataset
+        """
+
+        if "expect_column_values_to_be_between" not in self.excluded_expectations:
+            min_value = dataset.expect_column_min_to_be_between(
+                column,
+                min_value=None,
+                max_value=None,
+                parse_strings_as_datetimes=True,
+                result_format="SUMMARY",
+            ).result["observed_value"]
+
+            if min_value is not None:
+                try:
+                    min_value = min_value + datetime.timedelta(days=-365)
```

Comment:
Why are the bounds for datetime expectations one year before and one year after the min & max?

Reply:
Great catch! Holdover from an earlier version.